server/src/main/java/org/opensearch/transport/TransportService.java

```diff
@@ -210,7 +210,7 @@ public TransportService(
         setTracerLogInclude(TransportSettings.TRACE_LOG_INCLUDE_SETTING.get(settings));
         setTracerLogExclude(TransportSettings.TRACE_LOG_EXCLUDE_SETTING.get(settings));
         tracerLog = Loggers.getLogger(logger, ".tracer");
-        taskManager = createTaskManager(settings, threadPool, taskHeaders);
+        taskManager = createTaskManager(settings, clusterSettings, threadPool, taskHeaders);
```

Comment:
Nitpick, but since there is already a branch on whether clusterSettings is null, I'd just do the assignment there instead of creating a method (and add an else branch to assign taskManager if there are no cluster settings).

Reply:
createTaskManager method will be required as tests override this method to provide custom functionality. 

```
@Override
    protected TaskManager createTaskManager(
        Settings settings,
        ClusterSettings clusterSettings,
        ThreadPool threadPool,
        Set<String> taskHeaders
    ) {
        if (MockTaskManager.USE_MOCK_TASK_MANAGER_SETTING.get(settings)) {
            return new MockTaskManager(settings, threadPool, taskHeaders);
        } else {
            return super.createTaskManager(settings, clusterSettings, threadPool, taskHeaders);
        }
    }
```